api/Service/UserService.cs

```diff
@@ -1,66 +1,83 @@
 using YonderfulApi.Data;
-using YonderfulApi.Models;
-using api.Service;
 
-namespace YonderfulApi.Service
+namespace api.Service
 {
-    public class UserService : IUserService
+  public class UserService : IUserService
+  {
+    private HashingManager hashing = new HashingManager();
+    private readonly DataContext _context;
+    public UserService(DataContext context)
     {
-        private HashingManager hashing = new HashingManager();
-        private readonly DataContext _context;
-        public UserService(DataContext context)
-        {
-            _context = context;
-        }
+      _context = context;
+    }
 
-        public async Task<User> GetUserById(int userId)
-        {
-            var user = await _context.Users.FindAsync(userId);
-            return user;
-        }
+    public async Task<User> GetUserById(int userId)
+    {
+      var user = await _context.Users.FindAsync(userId);
+      return user;
+    }
 
-        public async Task<User> GetUserByEmail(string email)
-        {
-            var user = await _context.Users.FirstOrDefaultAsync(u => u.Email == email);
-            return user;
-        }
+    public async Task<User> GetUserByEmail(string email)
+    {
+      var user = await _context.Users.FirstOrDefaultAsync(u => u.Email == email);
+      return user;
+    }
 
-        public async Task<IList<User>> GetUserList()
-        {
-            var userList = await _context.Users.ToListAsync();
-            return userList;
-        }
+    public async Task<IList<User>> GetUserList()
+    {
+      var userList = await _context.Users.ToListAsync();
+      return userList;
+    }
 
-        public async Task<User> PostUser(User user)
-        {
-            var newUser = new User
-            {
-                Name = user.Name,
-                Password = hashing.HashToString(user.Password),
-                Email = user.Email
-            };
+    public async Task<User> PostUser(User user)
+    {
+      var newUser = new User
+      {
+        Name = user.Name,
+        Password = hashing.HashToString(user.Password),
+        Email = user.Email
+      };
 
-            _context.Users.Add(newUser);
-            await _context.SaveChangesAsync();
-            return newUser;
-        }
+      _context.Users.Add(newUser);
+      await _context.SaveChangesAsync();
+      return newUser;
+    }
+
+    public async Task<bool> DeleteUser(int id)
+    {
+      var user = await _context.Users.FindAsync(id);
+      if (user == null) return false;
+
+      _context.Users.Remove(user);
+      return await _context.SaveChangesAsync() > 0;
+    }
 
-        public async Task<bool> DeleteUser(int id)
-        {
-            var user = await _context.Users.FindAsync(id);
-            if (user == null) return false;
+    private async Task<bool> UserExists(string email)
+    {
+      return await _context.Users.AnyAsync(u => u.Email.ToLower() == email.ToLower());
+    }
+
+    public async Task<User> PutUser(User user)
```

Comment:
Here only the
name, phone and profession should be updated, since these are the only editable fields by the user on the my profile page.